sdk/go/common/workspace/plugins.go

```diff
@@ -720,6 +721,11 @@ func (info PluginInfo) installLock() (unlock func(), err error) {
 	}, nil
 }
 
+// Install installs a plugin's tarball into the cache. See InstallWithContext for details.
+func (info PluginInfo) Install(tgz io.ReadCloser, reinstall bool) error {
```

Comment:
It is very lame that we class this stuff as public api in sdk/go. I can't imagine anything but the engine cares about it.